File: Contests/Contest 80/819.Most-Common-Word/solution.cpp
#include <bits/stdc++.h>
using namespace std;


class Solution {
public:
    string mostCommonWord(string p, vector<string>& banned) {
        unordered_set<string> ban(banned.begin(), banned.end());
        unordered_map<string, int> count;
        paragraph += '.';
        int n = paragraph.size(), freq = 0;
        string ans, word;
        for (int i = 0; i < n; ++i) {
            if (isalpha(paragraph[i])) {
                word += tolower(paragraph[i]);
            }
            else if (!word.empty()) {
                if (ban.find(word) == ban.end() && ++count[word] > freq) {
                    freq = count[word];
                    ans = word;
                }
                word = "";
            }
        }
        return ans;
    }
};


int main() {
    Solution s;
    string paragraph = "hi hi ok.";
    vector<string> banned;
    cout << s.mostCommonWord(paragraph, banned) << endl;
    return 0;
}

File: Contests/Contest 80/819.Most-Common-Word/solution.py
import collections

class Solution(object):

    def mostCommonWord(self, paragraph, banned):
        banset = set(banned)
        count = collections.Counter(
            word.strip("!?',;.") for word in paragraph.lower().split())

        ans, best = '', 0
        for word in count:
            if count[word] > best and word not in banset:
                ans, best = word, count[word]

        return ans

############ test case ###########
s = Solution()
print s.mostCommonWord("sister's ball", [])
